src/types.ts

```diff
@@ -70,3 +70,47 @@ export interface UserSettings {
 export function isStreetViewImagery(x: string): x is "google" | "bing" {
   return x == "google" || x == "bing";
 }
+
+export type Mode =
+  | "edit-attribute"
+  | "edit-geometry"
+  | "route"
+  | "point"
+  | "free-polygon"
+  | "snap-polygon"
+  | "split-route"
+  | "street-view";
+
+export const MapEvents = [
+  "click",
+  "dblclick",
+  "mousemove",
+  "mousedown",
+  "mouseup",
+  "mouseout",
+  "dragstart",
+];
+
+export type MapEvent = (typeof MapEvents)[number];
+
+export const DocumentEvents = ["click", "keyPress", "keyUp", "keyDown"];
+
+export type DocumentEvent = (typeof DocumentEvents)[number];
+
+export interface EventHandler {
```

Comment:
Do we need the interface separate from the base class? I think we could remove this and just use the class everywhere, including for the type when passing stuff into each mode

Reply:
Most likely